utils/utils.c

```diff
@@ -283,14 +284,118 @@ void fetchGlfsVolServerFromEnv()
   LOG("mgmt", GB_LOG_INFO, "Block Hosting Volfile Server Set to: %s", gbConf.volServer);
 }
 
+
+static int
+glusterLogrotateConfigSet(char *logDir)
+{
+  char *buf = NULL, *line = NULL, *p, *dom = NULL;
+  int ret, m, len;
+  size_t n;
+  FILE *fp;
+
+
+  if (gbCtx == GB_CLI_MODE) {
+    dom = "cli";
+  } else if (gbCtx == GB_DAEMON_MODE) {
+    dom = "mgmt";
+  }
+
+  if (!dom) {
+    return -EINVAL;
+  }
+
+  fp = fopen(GB_LOGROTATE_PATH, "r+");
+  if (fp == NULL) {
+    ret = -errno;
+    LOG(dom, GB_LOG_ERROR, "Failed to open file '%s', %s", GB_LOGROTATE_PATH,
+        strerror (errno));
+    return ret;
+  }
+
+  ret = fseek(fp, 0L, SEEK_END);
+  if (ret == -1) {
+    ret = -errno;
+    LOG(dom, GB_LOG_ERROR, "Failed to seek file '%s', %s", GB_LOGROTATE_PATH,
+        strerror (errno));
+    goto error;
+  }
+
+  len = ftell(fp);
+  if (len == -1) {
+    ret = -errno;
+    LOG(dom, GB_LOG_ERROR, "Failed to get the length of file '%s', %s",
+        GB_LOGROTATE_PATH, strerror (errno));
+    goto error;
+  }
+
+  /* to make sure we have enough size */
+  len += strlen(logDir) + 1;
+  if (GB_ALLOC_N(buf, len) < 0) {
+    ret = -ENOMEM;
+    goto error;
+  }
+
+  p = buf;
+  fseek(fp, 0L, SEEK_SET);
+  while ((m = getline(&line, &n, fp)) != -1) {
+    if (strstr(line, "*.log") && strchr(line, '{')) {
+      m = sprintf(p, "%s/*.log {\n", logDir);
+    } else {
+      m = sprintf(p, "%s", line);
+    }
+    if (m < 0) {
+      ret = m;
+      goto error;
+    }
+
+    p += m;
+  }
+  *p = '\0';
+  len = p - buf;
+
+  fseek(fp, 0L, SEEK_SET);
+  truncate(GB_LOGROTATE_PATH, 0L);
+  ret = fwrite(buf, 1, len, fp);
+  if (ret != len) {
+    LOG(dom, GB_LOG_ERROR, "Failed to update '%s', %s", GB_LOGROTATE_PATH,
+        strerror (errno));
+    goto error;
+  }
+
+  ret = 0;
+ error:
+  if (fp) {
+    fclose(fp);
+  }
+  GB_FREE(buf);
+  GB_FREE(line);
+  return ret;
+}
+
+
 static int
 initLogDirAndFiles(char *newLogDir)
 {
   char *logDir = NULL;
   char *tmpLogDir = NULL;
+  char *dom = NULL;
   int ret = 0;
+  bool def = false;
+  int logLevel;
 
 
+  if (gbCtx == GB_CLI_MODE) {
+    dom = "cli";
+    logLevel = GB_LOG_DEBUG;
+  } else if (gbCtx == GB_DAEMON_MODE) {
+    dom = "mgmt";
+    logLevel = GB_LOG_CRIT;
```

Comment:
@lxbsz why setting logLevel here ?

Reply:
ignore, I got it.

Reply:
@lxbsz doesn't it makesense to log about LOGDIR change in CRIT level for cli too ?

Reply:
@pkalever 
IMO the cli is one short process in a very short time, it's okay to keep it as DEBUG.
Make sense ?

Reply:
If you want to change it to CRIT, I can update it right now.

Reply:
But then, if we do that, I think on every cli call we will log it :-D

Lets live with it for now, no changes. Thanks!